File: car-lease-service/src/test/java/org/jesperancinha/car/lease/repository/UserRepositoryTest.java
package org.jesperancinha.car.lease.repository;

import org.jesperancinha.car.lease.model.User;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
class UserRepositoryTest {

    @Autowired
    private UserRepository userRepository;

    @Test
    public void testSaveCar_whenGoodCar_thenSaveCar() {
        final var username = User.builder().username("Joao").build();
        final var usernameSave = userRepository.save(username);
        assertThat(usernameSave.getId()).isNotNull();
        assertThat(usernameSave.getUsername()).isEqualTo("Joao");
    }

}

File: car-lease-service/src/test/kotlin/org/jesperancinha/car/lease/repository/UserRepositoryTest.kt
package org.jesperancinha.car.lease.repository

import org.jesperancinha.car.lease.model.User
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest

@DataJpaTest
internal class UserRepositoryTest {
    @Autowired
    private val userRepository: UserRepository? = null
    @Test
    fun testSaveCar_whenGoodCar_thenSaveCar() {
        val username: Unit = User.builder().username("Joao").build()
        val usernameSave = userRepository!!.save<User>(username)
        assertThat(usernameSave.getId()).isNotNull()
        assertThat(usernameSave.getUsername()).isEqualTo("Joao")
    }
}